tree/dataframe/src/RDFInterfaceUtils.cxx

```diff
@@ -753,6 +726,29 @@ ColumnNames_t GetValidatedColumnNames(RLoopManager &lm, const unsigned int nColu
    return selectedColumns;
 }
 
+std::vector<std::string> GetValidatedArgTypes(const ColumnNames_t &colNames, const RBookedCustomColumns &customColumns,
+                                              TTree *tree, RDataSource *ds, const std::string &context,
+                                              bool vector2rvec)
+{
+   auto toCheckedArgType = [&](const std::string &c) {
+      RDFDetail::RCustomColumnBase *customCol =
+         customColumns.HasName(c) ? customColumns.GetColumns().at(c).get() : nullptr;
+      const auto colType = ColumnName2ColumnTypeName(c, tree, ds, customCol, vector2rvec);
+      if (colType.rfind("CLING_UNKNOWN_TYPE", 0) == 0) { // the interpreter does not know this type
+         const auto msg =
+            "The type of custom column \"" + c + "\" (" + colType.substr(19) +
```

Comment:
`substr(19)` puzzles me. Why is this a fixed number?

Reply:
it's `std::string("CLING_UNKNOWN_TYPE_").size()`. what would you suggest?

Reply:
Oh, now I see it. Ok, it's safe here.

Reply:
... unless the string changes at some point. If that might happen, you could do this:
```
      const auto unknownType = "CLING_UNKNOWN_TYPE";
      if (rfind(unknownType)...
            + unknownType
```
It's unlikely, though, that cling type names change, I assume.

Reply:
`CLING_UNKNOWN_TYPE` is set by RDF itself, so it won't change unless RDF decides it should change. nevertheless I don't like the magic string floating around -- but I don't know of a nicer solution that would not complicate things further. I hope the unit test I added is enough to catch things going out of sync

Reply:
ok _now_ the test will definitely notice if the diagnostic changes :D